pkg/applicationserver/io/pubsub/grpc_pubsub.go

```diff
@@ -68,7 +69,7 @@ func (ps *PubSub) List(ctx context.Context, req *ttnpb.ListApplicationPubSubsReq
 
 // Set implements ttnpb.ApplicationPubSubRegistryServer.
 func (ps *PubSub) Set(ctx context.Context, req *ttnpb.SetApplicationPubSubRequest) (*ttnpb.ApplicationPubSub, error) {
-	if err := rights.RequireApplication(ctx, req.ApplicationIdentifiers, ttnpb.RIGHT_APPLICATION_TRAFFIC_READ); err != nil {
+	if err := rights.RequireApplication(ctx, req.ApplicationIdentifiers, ttnpb.RIGHT_APPLICATION_TRAFFIC_DOWN_WRITE); err != nil {
```

Comment:
I think we need to assert both rights

Reply:
In both cases ?

Reply:
Yeah cause it implicitly ends up doing read/write, bypassing auth checks

@htdvisser what do you think?

Reply:
I think that in both the Webhooks and Pubsub we should require `RIGHT_APPLICATION_SETTINGS_BASIC` for managing the integrations. Or perhaps we could introduce a new right for managing application integrations

Reply:
Webhooks and pub/sub are strictly traffic related though.

How about a new right `TRAFFIC_ALL` that implies read/write?

Reply:
Yes, but adding/removing webhook/pubsub configurations is a management action. If I give an external platform an API key for reading/writing traffic, it's strange that they would be able to read/update/delete my other webhook/pubsub configurations.

Reply:
Right. In that case we need to assert `SETTINGS_BASIC` as well. We must still assert `TRAFFIC` rights, because you should not be able to read/write all traffic with just settings rights.

I'm also okay with a new right instead of asserting 4 rights or 1 + 1 implied traffic all right.